gsa/cigar.go

```diff
@@ -13,6 +18,14 @@ package gsa
 //      The edit operations the CIGAR string describes.
 func CigarToEdits(cigar string) (edits string) {
 	edits = ""
+	for pos, char := range cigar {
+		char := string(char)
+
+		//check if string is a number
+		if num, err := strconv.Atoi(char); err == nil {
+			edits += strings.Repeat(string(cigar[pos+1]), num)
+		}
+	}
```

Comment:
This looks mighty weird. I can _almost_ see what you were thinking. You are running through the characters, and every time you can interpret one as a number, you grab it and then consider the following character the operation.

The mistake I've seen the most in this exercise is assuming that every second index has a number (`pos = 0, 2, 4, ...` and that the edit operation is on the positions after that `pos = 1, 3, 5, ...`). This is not correct, as there are numbers known to be larger than single-digit. So don't do that. But what you are doing isn't any better. How does this code handle the cigar "11M21D"? It discovers that 1 is a number and records 1 "1" operation. Then it discovers that 1 is a number and records 1 "M" operation. Then it gets an error for "M" and skips that and then record two "1" operations. You can see where this is going.